storage_service/locations/models/s3.py

```diff
@@ -76,7 +76,13 @@ def resource(self):
         return self._resource
 
     def _ensure_bucket_exists(self):
-        self.client.create_bucket(Bucket=self._bucket_name())
+        try:
+            self.client.head_bucket(Bucket=self._bucket_name())
+        except:
```

Comment:
```suggestion
        except self.client.exceptions.ClientError:
```
Nice! I've offered a small suggestion for the except. While there's little that can go wrong here, we should try and capture specific exceptions just in case, so we don't mask what those might be. 

I think this looks like the right syntax from these docs here but you might need to try it out: https://boto3.amazonaws.com/v1/documentation/api/latest/guide/migrations3.html#accessing-a-bucket.

Reply:
Thanks @ross-spencer :)